File: LeetCode/JavaScript/#136 Single Number.js
/**
 * 
 * @authors Your Name (you@example.org)
 * @date    2017-04-04 18:19:05
 * @version $Id$
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    return nums.reduce((a,b)=>{return a^b;});
};

File: LeetCode/Python/#136 Single Number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return reduce(lambda x,y:x^y,nums)




#·½·¨¶ş
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sum(list(set(nums)))*2-sum(nums)
